File: 0035-search-insert-position/solution.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int len = nums.size();
        int j;
        for(int i=0; i<len; i++){
            if(nums[i]==target)
                return i;
        }
        for(j=0; j<len; j++){
            if(nums[j]>target)     //index to insert
                return j;
        }
        return j;
    }
};

File: 0035-search-insert-position/solution.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        n = len(nums)
        low = 0
        high = n-1
        while low<=high:
            mid = (low+high)//2
            if nums[mid]==target:
                return mid
            elif nums[mid]>target:
                high = mid-1
            else:
                low = mid+1
        '''
        if nums[mid]<target:
            mid+=1
        if mid<0:
            return 0
        return mid
        '''
        return low
